src/pvq_encoder.c

```diff
@@ -283,7 +283,7 @@ typedef struct {
 } pvq_search_item;
 
 int items_compare(pvq_search_item *a, pvq_search_item *b) {
-  return a->k - b->k;
+  return a->k == b->k ? a->gain - b->gain : a->k - b->k;
```

Comment:
Please put a comment above this line explaining that the gain sorting is done to make the sort stable